### PR TITLE
uncommented the required permission

### LINKED ISSUE #397: Misconfigured cloud-controller-manager.yml (HA deployment that uses daemonset)
After deploying got errors:
```
E1119 10:01:56.059612       1 leaderelection.go:325] error retrieving resource lock kube-system/cloud-controller-manager: leases.coordination.k8s.io "cloud-controller-manager" is forbidden: User "system:serviceaccount:kube-system:cloud-controller-manager" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "kube-system"
```

So need to add more permissions to make it works:
```
- apiGroups:
  - "coordination.k8s.io"
  resources:
  - leases
  verbs:
  - create
  - get
  - list
  - watch
  - update
```

### PR DESCRIPTION
**Pull request by cpanato**
- uncommented the required permission and made a copy of the file and added a -dev suffix

Fixes #397

/assign @timoreimann

### CODE REVIEW

File: releases/v0.1.36-dev.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
```
Comment (timoreimann) [inline releases/v0.1.36-dev.yml:1]: Would you remind renaming the file name to just `dev.yaml`? Our to be updated release pipeline / job should be responsible for picking a proper name at release time.

**Review by timoreimann — COMMENTED** [review]
One small gotcha, otherwise good.

File: releases/v0.1.36-dev.yml (hunk 1 above)
Comment (cpanato) [reply releases/v0.1.36-dev.yml]: done @timoreimann, thanks for the review

**Comment by timoreimann** [comment]
I have filed #415 to track updating our release pipeline to account for the new `dev.yaml` file.

**Pull request merged**